### PR TITLE
Clean up some dead code and fix some typos.

### PR DESCRIPTION
**Pull request by brendandburns**
@janetkuo @lavalamp

### CODE REVIEW

**Comment by janetkuo** [comment]
Travis complains that you need to run hack/update-codecgen.sh.

**Comment by brendandburns** [comment]
@janetkuo 
Done.  ptal.

thanks!
--brendan

File: hack/update-codecgen.sh [hunk 1]
```diff
@@ -54,7 +54,7 @@ function depends {
   file=${generated_files[$1]//\.generated\.go/.go}
   deps=$(go list -f "{{.Deps}}" ${file} | tr "[" " " | tr "]" " ")
   inputfile=${generated_files[$2]//\.generated\.go/.go}
-  fullpath=$(readlink -f ${generated_files[$2]//\.generated\.go/.go})
+  fullpath=${generated_files[$2]//\.generated\.go/.go}
```
Comment (janetkuo) [inline hack/update-codecgen.sh]: I'm not sure why change this?

Comment (brendandburns) [inline hack/update-codecgen.sh]: oh, right, this is broken on OS X, but I should revert it here and do it in a different PR.

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Comment by brendandburns** [comment]
@k8s-bot unit test this please
